src/OrbitGl/GraphTrack.cpp

```diff
@@ -82,8 +82,7 @@ void GraphTrack<Dimension>::Draw(Batcher& batcher, TextRenderer& text_renderer,
 template <size_t Dimension>
 void GraphTrack<Dimension>::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t max_tick,
                                              PickingMode picking_mode, float z_offset) {
-  float track_width = viewport_->GetVisibleWorldWidth();
-  SetSize(track_width, GetHeight());
+  SetSize(GetSize()[0], GetHeight());
```

Comment:
Why is this SetSize still needed?

Reply:
Thanks for pointing out this!

I spent some time to figure out this issue yesterday. If not update the track height here, there will be some issue when we expand the major page fault subtrack but collapse the page fault track. Specifically, it looks like: https://screen/8T3uHPNe3fseDGJ

This is because, now we change to update track size and position in `TimeGraph::DrawTracks` before drawing tracks.
However for the subtracks, for instance, the major page fault subtrack, their size and position are updated in the parent track. 

When the parent track is expanded, everything works fine. But if it is collapsed, the subtrack size won't be updated, as illustrated in the following code.

```cpp
void PageFaultsTrack::Draw(Batcher& batcher, TextRenderer& text_renderer,
                           uint64_t current_mouse_time_ns, PickingMode picking_mode,
                           float z_offset) {
 ...

  UpdatePositionOfSubtracks();

  Track::Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode, z_offset);

  if (collapse_toggle_->IsCollapsed()) return;

  if (!major_page_faults_track_->IsEmpty()) {
    major_page_faults_track_->SetSize(size_[0], major_page_faults_track_->GetHeight());
    major_page_faults_track_->Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode,
                                   z_offset);
  }

....
}
```

However, the collapsed page fault track would like to show a collapsed  major page fault subtrack. But here the size of the track is not updated.

Reply:
I will change the code to update the subtrack size in the parent track's draw method. Namely,

```
void PageFaultsTrack::Draw(Batcher& batcher, TextRenderer& text_renderer,
                           uint64_t current_mouse_time_ns, PickingMode picking_mode,
                           float z_offset) {
 ...

  UpdatePositionOfSubtracks();

  Track::Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode, z_offset);

  if (!major_page_faults_track_->IsEmpty()) {
    major_page_faults_track_->SetSize(size_[0], major_page_faults_track_->GetHeight());
  }

  if (collapse_toggle_->IsCollapsed()) return;

  if (!major_page_faults_track_->IsEmpty()) {
    major_page_faults_track_->Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode,
                                   z_offset);
  }

....
}

```

And I will do the similar change for the gpu track to always update the size of gpu submission subtrack in `GpuTrack::Draw`

Reply:
> Really happy to see this PR. In the second commit you are mentioning that you are moving a method to TrackManager and you aren't (which I agree). You should modify commit msg! :)
> 
> I would appreciate if you merge this after my text-box removal PR :)

Sure. I will wait for your textbox goodbye PR! BTW, I like the PR name ;-p

Reply:
Thanks for the detailed explanation and for the fixup!